src/communication/xcConnection.ts

```diff
@@ -34,7 +29,9 @@ export class DefaultConnection implements Connection {
             log.debug("getModel request failed");
             log.debug(err);
         };
-        session.init(openListener, errorListener);
+        let closeListener = (_: CloseEvent) => {
```

Comment:
then we should call the listener with an error when a reconnection happens.... no? 
i think we should clarify the mechanics of error x reconnection, and the tests (that are missing) would help clarifying how things should work.

is a disconnection an error or not?
when it happens is the error callback called or not?
when we can't reconnect after the given interval, is the error callback called or not?
or it is called upon each reconnection?

I my opinion we shouldn't need any extra callbacks, and we should need a new kind of error, something in the lines of "I tried to reconnect N times as you asked but it didn't work, Error"
Besides, it should be a API level config OR parameters added to all methods in the API (ugly). 

I don't see the point of adding it to some methods and not to others.

@nicolaserny  what's your take on that?

BTW, I don't think we need to wait for this PR to finish to release a new version.